.github/workflows/auto-upgrade-certified-connectors-cdk.yml

```diff
@@ -0,0 +1,213 @@
+        with:
+          owner: "airbytehq"
+          repositories: "airbyte"
+          app-id: ${{ secrets.OCTAVIA_BOT_APP_ID }}
+          private-key: ${{ secrets.OCTAVIA_BOT_PRIVATE_KEY }}
+      - name: Get GitHub App User ID
+        id: get-user-id
+        run: echo "user-id=$(gh api "/users/${{ steps.app-token.outputs.app-slug }}[bot]" --jq .id)" >> "$GITHUB_OUTPUT"
+        env:
+          GH_TOKEN: ${{ steps.app-token.outputs.token }}
+      - run: |
+          git config --global user.name '${{ steps.app-token.outputs.app-slug }}[bot]'
+          git config --global user.email '${{ steps.get-user-id.outputs.user-id }}+${{ steps.app-token.outputs.app-slug }}[bot]@users.noreply.github.com'
+
+      - name: Checkout Airbyte
+        uses: actions/checkout@08eba0b27e820071cde6df949e0beb9ba4906955 # v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install yq
+        run: sudo snap install yq
+
+      - name: Setup Java
+        uses: actions/setup-java@c5195efecf7bdfc987ee8bae7a71cb8b11521c00 # v4
+        with:
+          distribution: "zulu"
+          java-version: "21"
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@80e941e61874822d2a89974089c4915748e8f4b7 # v4
+
+      - name: Run upgradeCdk for ${{ matrix.connector }}
+        id: upgrade-cdk
+        run: |
+          ./gradlew :airbyte-integrations:connectors:${{ matrix.connector }}:upgradeCdk
+          # --quiet disables all of gradle's normal logging, so we only get the CDK version number
+          new_cdk_version=$(./gradlew :airbyte-integrations:connectors:${{ matrix.connector }}:getCdkVersion --quiet)
+          echo "new_cdk_version=$new_cdk_version" >> $GITHUB_OUTPUT
+
+      - name: Check for changes
+        id: check-changes
+        if: steps.upgrade-cdk.outputs.exit_code == '0'
+        run: |
+          if git diff --quiet; then
+            echo "has_changes=false" >> $GITHUB_OUTPUT
+            echo "No changes detected for ${{ matrix.connector }}"
+          else
+            echo "has_changes=true" >> $GITHUB_OUTPUT
+            echo "Changes detected for ${{ matrix.connector }}"
+          fi
+
+      - name: Bump connector version
+        id: bump-version
+        if: steps.check-changes.outputs.has_changes == 'true'
+        run: |
+          metadata_file="airbyte-integrations/connectors/${{ matrix.connector }}/metadata.yaml"
+
+          # Extract current version
+          current_version=$(yq '.data.dockerImageTag' "$metadata_file")
+          echo "Current version: $current_version"
+
+          # Parse version components
+          IFS='.' read -r major minor patch <<< "$current_version"
+
+          # Increment patch version
+          new_patch=$((patch + 1))
+          new_version="${major}.${minor}.${new_patch}"
+          echo "New version: $new_version"
+
+          # Update metadata.yaml
+          yq --inplace --prettyPrint ".data.dockerImageTag = \"${new_version}\"" "$metadata_file"
```

Comment:
yq does some annoying formatting changes. I could be convinced to use `sed` instead. E.g.
* https://github.com/airbytehq/airbyte/pull/67070/files
* https://github.com/airbytehq/airbyte/pull/67066/files
* https://github.com/airbytehq/airbyte/pull/67063/files

Reply:
hm would any of our existing format script reformat these back?

Reply:
I'm switching to sed. yq is introducing changes that actually fail our format checks >.> https://github.com/airbytehq/airbyte-enterprise/pull/275/files

yq-generated diff:
<img width="316" height="70" alt="Image" src="https://github.com/user-attachments/assets/aa8a10e6-9701-43eb-8856-80068f3e201e" />

formatter wants to change it back 🤦 
<img width="260" height="54" alt="Image" src="https://github.com/user-attachments/assets/0fdcc4e1-59b1-4ff7-83eb-54b58ad6b2ee" />

Reply:
sounds good